care/emr/api/viewsets/invoice.py

```
@@ -131,41 +131,45 @@ def authorize_update(self, request_obj, model_instance):
-                        id__in=instance.charge_items,
-                    ).update(
-                        status=ChargeItemStatusOptions.paid.value, paid_invoice=instance
-                    )
-            super().perform_update(instance)
+            with transaction.atomic():
+                old_invoice = Invoice.objects.get(id=instance.id)
+                if old_invoice.status != instance.status:
+                    if instance.status in INVOICE_CANCELLED_STATUS:
+                        raise ValidationError(
+                            "Call the cancel invoice API to cancel the invoice"
+                        )
```

Ruff (0.14.4)

Avoid specifying long messages outside the exception class

(TRY003)